src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -140,20 +147,60 @@ const processTxForWallet = async (
             ratio: account.depositFeeRatio,
           })
 
-          const converter = DisplayCurrencyConverter(displayCurrencyPerSat)
-          const amountDisplayCurrency = converter.fromSats(sats)
-          const feeDisplayCurrency = converter.fromSats(fee)
+          const receivedBtc = paymentAmountFromNumber({
+            amount: sats,
+            currency: WalletCurrency.Btc,
+          })
+          if (receivedBtc instanceof Error) return receivedBtc
 
-          const result = await ledger.addOnChainTxReceive({
-            walletId: wallet.id,
-            walletCurrency: wallet.currency,
-            txHash: tx.rawTx.txHash,
-            sats,
-            fee,
-            amountDisplayCurrency,
+          const feeBtc = paymentAmountFromNumber({
+            amount: fee,
+            currency: WalletCurrency.Btc,
+          })
+          if (feeBtc instanceof Error) return feeBtc
+
+          const walletAddressReceiver = await WalletAddressReceiver({
```

Comment:
I don't know if this is from this PR or already in the code base but an `async` constructor function is very suspicious to me.

Reply:
I pretty much modeled this from the `WalletInvoiceReceiver`. Async in both cases is because we call conversion methods from the dealer async. Could potentially be a code smell 🤔